projects/npm-tools/packages/npm-scripts/test/utils/tweakWebpackOptions.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const path = require('path');
+const {
+	container: {ModuleFederationPlugin},
+} = require('webpack');
+
+const getMergedConfig = require('../../src/utils/getMergedConfig');
+const tweakWebpackConfig = require('../../src/utils/tweakWebpackConfig');
+
+const BABEL_CONFIG = getMergedConfig('babel');
+
+describe('tweakWebpackConfig() without federation', () => {
+	it('returns unchanged config when it doesn\'t include a "module" field', () => {
+		const config = {};
+
+		expect(tweakWebpackConfig(config)).toEqual(config);
+	});
+
+	it("returns unchanged config when module configs don't include rules", () => {
+		const config = {module: {}};
+
+		expect(tweakWebpackConfig(config)).toEqual(config);
+	});
+
+	it('returns unchanged rules when they don\'t include a "use" field', () => {
+		const config = {
+			module: {
+				rules: [
+					{
+						exclude: /node_modules/,
+						test: /\.(js|jsx)$/,
+					},
+				],
+			},
+		};
+
+		expect(tweakWebpackConfig(config)).toEqual(config);
+	});
+
+	it('returns unchanged rules when the loader is not the babel-loader', () => {
+		const config = {
+			module: {
+				rules: [
+					{
+						test: /\.(js|jsx)$/,
+						use: 'ts-loader',
+					},
+					{
+						test: /\.(js|jsx)$/,
+						use: ['ts-loader'],
+					},
+				],
+			},
+		};
+
+		expect(tweakWebpackConfig(config)).toEqual(config);
+	});
+
+	it('merges babel options when the loader is the babel-loader', () => {
+		const config = {
+			module: {
+				rules: [
+					{
+						test: /\.(js|jsx)$/,
+						use: 'babel-loader',
+					},
+					{
+						test: /\.(js|jsx)$/,
+						use: [
+							{
+								loader: 'babel-loader',
+								options: {
+									include: 'bruno',
+								},
+							},
+						],
+					},
+				],
+			},
+		};
+
+		const merged = tweakWebpackConfig(config);
+
+		expect(merged).toEqual({
+			module: {
+				rules: [
+					{
+						test: /\.(js|jsx)$/,
+						use: {
+							loader: 'babel-loader',
+							options: BABEL_CONFIG,
+						},
+					},
+					{
+						test: /\.(js|jsx)$/,
+						use: [
+							{
+								loader: 'babel-loader',
+								options: {
+									...BABEL_CONFIG,
+									include: 'bruno',
+								},
+							},
+						],
+					},
+				],
+			},
+		});
+	});
+});
+
+describe('tweakWebpackConfig() with federation', () => {
+	let savedCwd;
+
+	beforeEach(() => {
+		savedCwd = process.cwd();
+
+		process.chdir(
+			path.join(
+				__dirname,
+				'..',
+				'..',
+				'__fixtures__',
+				'utils',
+				'tweakWebpackConfig'
+			)
+		);
+	});
+
+	afterEach(() => {
+		fs.rmdirSync('build', {recursive: true});
+
+		process.chdir(savedCwd);
+	});
```

Comment:
It's probably best not to write files inside the repo during a test run. This is something that I pointed out in Bryce's PR [over here](https://github.com/liferay/liferay-frontend-projects/pull/245#discussion_r527577883):

> If the tests crash while running, you'll wind up with a dirty work tree here. That's why in other tests in `npm-scripts` we create a temporary directory and do all our tests in there.
>
> If you wanted to take that approach, you could find examples [like this one](https://github.com/liferay/liferay-frontend-projects/blob/542e42c5b24897a1e63016afae7c44e7351e526e/projects/npm-tools/packages/npm-scripts/test/utils/expandGlobs.js) by searching for `os.tmpdir()`.
>
> You don't need to clean up the files in that case (the OS will clean the temp dirs), so you can avoid the dependency on `rimraf` too, as well as the time spent erasing the files.